sql/migrations/0025-orc-lifecycles.sql

```diff
@@ -0,0 +1,29 @@
+SELECT run_migration(25, $$
+
+create table workflow_lifecycle (
+	id          uuid        primary key default gen_random_uuid(),
+	name        text        not null,
+	version     int         not null default 1,
+	created     timestamptz not null default current_timestamp,
+	updated     timestamptz not null default current_timestamp,
+	deactivated timestamptz,
+	locked      bool        not null default false,
+	role_id     uuid        not null references device_role(id),
+	unique(name, version)
+);
+create unique index on workflow_lifecycle(name) where deactivated is null;
```

Comment:
With the two unique indices on this table, do I follow that:

* Only one lifecycle with a given name can be active at a given time
* There can be multiple versions of the lifecycle with a given name, but older ones must be deactivated